package/kernel/linux/modules/netsupport.mk

```diff
@@ -777,6 +777,37 @@ endef
 $(eval $(call KernelPackage,sched))
 
 
+define KernelPackage/tcp-bbr
+  SUBMENU:=$(NETWORK_SUPPORT_MENU)
+  TITLE:=BBR TCP congestion control
+  DEPENDS:=@!LINUX_3_18 @!LINUX_4_1 @!LINUX_4_4 +LINUX_4_9:kmod-sched
+  KCONFIG:= \
+	CONFIG_TCP_CONG_ADVANCED=y \
+	CONFIG_TCP_CONG_BBR
+  FILES:=$(LINUX_DIR)/net/ipv4/tcp_bbr.ko
+  AUTOLOAD:=$(call AutoLoad,74,tcp_bbr)
+endef
+
+define KernelPackage/tcp-bbr/description
+ Kernel module for BBR (Bottleneck Bandwidth and RTT) TCP congestion
+ control. It requires the fq ("Fair Queue") pacing packet scheduler.
+ For kernel 4.13+, TCP internal pacing is implemented as fallback.
+endef
+
+ifdef CONFIG_LINUX_4_9
```

Comment:
It checked kernel 4.9 only. We have to add new kernel checks when upgrading kernel, e.g. 4.14

Reply:
For k4.13+, BBR can use the TCP internal pacing implemented when fq qdisc is not available [1].
So, `net.core.default_qdisc=fq` is only included in `/etc/sysctl.d/12-tcp-bbr.conf` shipped for k4.9 and hence the check.

Users can choose to use fq qdisc for k4.14 by setting `net.core.default_qdisc=fq` in `/etc/sysctl.conf` if they really want to use it instead of TCP internal pacing.

This is also why `kmod-sched` is only selected for k4.9.
`DEPENDS:=@!LINUX_3_18 @!LINUX_4_1 @!LINUX_4_4 +LINUX_4_9:kmod-sched`

[1] https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=218af599fa635b107cfe10acf3249c4dfe5e4123